src/server/db_slice.cc

```diff
@@ -830,4 +839,19 @@ size_t DbSlice::EvictObjects(size_t memory_to_free, PrimeIterator it, DbTable* t
   return freed_memory_fun();
 };
 
+void DbSlice::RegisterWatchedKey(std::string_view key, ConnectionState::ExecInfo* exec_info) {
+  watched_keys_[key].push_back(exec_info);
```

Comment:
I am pretty sure that watch works on keys for a single DbTable (i.e. in the context of SELECT) and if you select another DB it won't abort a transaction.

Reply:
discussed offline